uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/EntitlementCalculation.php

```diff
@@ -0,0 +1,406 @@
+   * period - the number of leaves taken on the previous period. It may be
+   * limited by the max number of days allowed to be carried forward for this
+   * calculation's absence type.
+   *
+   * @return int
+   */
+  public function getBroughtForward() {
+    if(!$this->shouldCalculateBroughtForward()) {
+      return 0;
+    }
+
+    $broughtForward = $this->getNumberOfDaysRemainingInThePreviousPeriod();
+    if($broughtForward > $this->absenceType->max_number_of_days_to_carry_forward) {
+      return $this->absenceType->max_number_of_days_to_carry_forward;
+    }
+
+    return $broughtForward;
+  }
+
+  /**
+   * Returns the Pro Rata for this calculation contract on the calculation
+   * period.
+   *
+   * The Pro Rata is given by:
+   * (no. working days to work / no. of working days) x contractual entitlement.
+   *
+   * The end result is rounded up to the nearest half day. Example:
+   *
+   * Number of working days to work: 212
+   * Number of working days: 253
+   * Contractual entitlement: 28
+   * Pro rata: (212 / 253) * 28 = 23.46 = 23.5 (rounded)
+   *
+   * @return float|int
+   */
+  public function getProRata() {
+    $numberOfWorkingDaysToWork = $this->getNumberOfWorkingDaysToWork();
+    $numberOfWorkingDays = $this->getNumberOfWorkingDays();
+    $contractualEntitlement = $this->getContractualEntitlement();
+
+    $proRata = ($numberOfWorkingDaysToWork / $numberOfWorkingDays) * $contractualEntitlement;
+    $roundedProRata = ceil($proRata * 2) / 2;
+
+    return $roundedProRata;
+  }
+
+  /**
+   * Returns the Contractual Entitlement for this calculation contract.
+   *
+   * The Contractual Entitlement is given by the leave amount set for this
+   * calculation contract and absence type + the number of public holidays in
+   * the period (if the leave settings on the contract allows this).
+   *
+   * @return int
+   */
+  public function getContractualEntitlement() {
+    $contractualEntitlement = 0;
+
+    $jobLeave = $this->getJobLeaveForAbsenceType();
+    if(!$jobLeave) {
+      return $contractualEntitlement;
+    }
+
+    $contractualEntitlement = $jobLeave['leave_amount'];
+    if($jobLeave['add_public_holidays']) {
+      $contractualEntitlement += $this->getNumberOfPublicHolidaysForPeriod();
+    }
+
+    return $contractualEntitlement;
+  }
+
+  /**
+   * Returns the calculated proposed entitlement.
+   *
+   * This is basically the Pro Rata + the number of days brought forward
+   *
+   * @return float|int
+   */
+  public function getProposedEntitlement() {
+    return $this->getProRata() + $this->getBroughtForward();
+  }
+
+  /**
+   * Returns the proposed entitlement for this AbsenceType and Contract on the
+   * previous period.
+   *
+   * @return int
+   */
+  public function getPreviousPeriodProposedEntitlement() {
+    $previousPeriodEntitlement = $this->getPreviousPeriodEntitlement();
+
+    if(!$previousPeriodEntitlement) {
```

Comment:
For short sentences that might be the case, but here it would only make the line longer and reduce readability